test/test_download.py

```
@@ -0,0 +1,92 @@
+
+    assert result == "temp_file.ext"
+    mock_bot.download_file.assert_called_once_with("path/to/file")
+    mock_path_open.assert_called_once_with("wb")
+    mock_path_open().write.assert_called_once_with(b"test content")
+
+def test_download_tg_missing_file_path(mocker):
+    """Test download_tg raises ValueError when file_path is missing."""
+    mock_file = mocker.MagicMock()
+    mock_file.file_path = None
+
+    with pytest.raises(ValueError, match="Telegram file path is missing."):
```

Ruff (0.15.5)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)